lib/argocd.libjsonnet

```diff
@@ -50,6 +50,9 @@ local ArgoApp(component, namespace, project='syn', secrets=true) = {
         prune: true,
         selfHeal: true,
       },
+      syncOptions: [
+        'ServerSideApply=true',
```

Comment:
Do we know if switching to server-side apply works without any issues for existing setups?

Reply:
I used it to apply the crds in the v0.8.0 upgrade, and used it in another component.  
But other thant that I have no widespread experience using this option.

Reply:
We might wan't to update components individually, as there are some issues with empty annotations or empty keys in templates.